src/main/java/org/radarcns/management/security/jwt/ManagementPortalJwtAccessTokenConverter.java

```diff
@@ -150,24 +150,15 @@ public OAuth2AccessToken enhance(OAuth2AccessToken accessToken,
                     new DefaultOAuth2AccessToken(refreshToken.getValue());
             // Refresh tokens do not expire unless explicitly of the right type
             refreshTokenToEnhance.setExpiration(null);
-
-            // I don't really get what is the point of doing this.
-            try {
-                Map<String, Object> claims = objectMapper
-                        .parseMap(JwtHelper.decode(refreshToken.getValue()).getClaims());
-                if (claims.containsKey(TOKEN_ID)) {
-                    refreshTokenToEnhance.setValue(claims.get(TOKEN_ID).toString());
-                }
```

Comment:
If I understand correctly, isn't this the code to allow a `TOKEN_ID` to be invalidated in the `approvalStore()`? The refresh token itself will change every update but this would allow a certain refresh token ID to be invalidated, both for the access token and the refresh token.

Reply:
Technically yes.It is the id. I am also little confused about this implementation. Main confusion is sometimes the value is set to a UUID which is the token-id and sometimes it is the encoded string. Based on some debugging it appears that the value we get at this point `JwtHelper.decode(refreshToken.getValue()).getClaims()` is a UUID and we cannot decode it as a JWT. so It always fails here. I will revert the changes to fit exactly what was there earlier. I will look into this again.